plugins/draw-tools.js

```diff
@@ -77,6 +77,109 @@ window.plugin.drawTools.setDrawColor = function(color) {
   });
 }
 
+// Easy create a draw from an IITC-plugin, allow passing an optional color parameter
+
+// common function used by folloing functions to trigger display on new drawn item
+window.plugin.drawTools.fireDraw = function(layer, layerType) {
+  map.fire('draw:created', {
+    layer: layer,
+    layerType: layerType
+  });
+}
+
+/**
+ * Create a polyline
+ * @param arrCoordArr Array of coordinates
+ * @param {string=} color (optional)
+ */
+window.plugin.drawTools.drawPolyline = function(arrCoordArr, color) {
```

Comment:
All following functions have the same structure, and perform the same action: temporary override currentcolor, and fire draw with params which are default for corresponding shape.

It may be better to refactor into single function.
But I tend to thing that something is wrong in the approach itself.

Reply:
e.g. if a script detects a Crosslink from a draw on a draw you might want to automatically highlight that by drawing the link in a different color, without changing the chosen drawcolor. 
_It's "put a red line here, whatever current color is."_ 
Link prolongation uses the functions too.

Putting that into a single function would not be nice either, as each has different parms. (Coords, arrCordsArr, Radius)

Reply:
> e.g. if a script detects a Crosslink from a draw on a draw you might want to automatically highlight that by drawing the link in a different color, without changing the chosen drawcolor.

I understand why one may want to use default color, or current color, or any specific.
But I do not understand for what purpose all these functions change default color.